presto-common/src/main/java/com/facebook/presto/common/function/SqlFunctionProperties.java

```diff
@@ -24,6 +24,7 @@
 {
     private final boolean parseDecimalLiteralAsDouble;
     private final boolean legacyRowFieldOrdinalAccessEnabled;
+    private final boolean legacyTypeCoercionWarningEnabled;
```

Comment:
It's not property to have this in presto-common. Presto common is only for extremely generic flags.

Reply:
Per offline discussion, we will keep this in `SqlFunctionProperties`. 

There are multiple options regarding this warning flag:

- Always check type and print warning regardless of whether legacy mode is on or off. This might have performance implication.
- Always use `FeaturesConfig#isLegacyDateTimestampToVarcharCoercion`. The downside is that all non-injected class would have to change constructor to adopt this. The fanout is non-trivial given there are a lot of static method trying to create `ExpressionAnalyzer`.
- Add session property to determine if we want to check for warning, but does not put it into `SqlFunctionProperties`. This means a lot of `ExpressionCompiler` related class (and their caller) needs to be changed, specifically, it changes all `CacheKey`-like classes. The fanout is also non-trivial.
- Add this session property into `SqlFunctionProperties`, which seems to be the cleanest way, and there are several session property already resides in this class.